### PR TITLE
HCX-54 Fix hid.Hydra#deleteEntityAddress method

### PR DESCRIPTION
**Pull request by dolfinus**

### CODE REVIEW

File: src/org/camunda/latera/bss/connectors/hid/hydra/Address.groovy [hunk 1]
```diff
@@ -562,6 +562,24 @@ trait Address {
...
+  private Boolean isSubjectAddress(def entityAddressId, def entityTypeId = null, def entityId = null) {
+    if (notEmpty(entityTypeId) || notEmpty(entityId)) {
+      return isSubject(entityTypeId ?: entityId)
+    } else {
+      LinkedHashMap objAddress = getObjAddress(entityAddressId)
+      if (objAddress) {
+        return false
+      }
+
+      LinkedHashMap subjAddress = getSubjAddress(entityAddressId)
```
Comment (sugarfree1) [inline src/org/camunda/latera/bss/connectors/hid/hydra/Address.groovy]: okay, be more optimistic and check subj address at first like so:
```
LinkedHashMap subjAddress = getSubjAddress(entityAddressId)

return isPresent(subjAddress) || isEmpty(getObjAddress(entityAddressId)) || null
```

you don't need ```if```'s

File: src/org/camunda/latera/bss/connectors/hid/hydra/Address.groovy [hunk 2]
```diff
@@ -584,7 +602,15 @@ trait Address {
       endDate            : null
     ], input)
 
-    Boolean isSubj = isSubject(params.entityTypeId ?: params.entityId)
+    Boolean isSubj = false
```
Comment (sugarfree1) [inline src/org/camunda/latera/bss/connectors/hid/hydra/Address.groovy]: name is not descriptive

File: src/org/camunda/latera/bss/connectors/hid/hydra/Address.groovy [hunk 3]
```diff
@@ -584,7 +602,15 @@ trait Address {
...
+
+    if (notEmpty(params.entityAddressId)) {
+      isSubj = isSubjectAddress(params.entityAddressId, params.entityTypeId, params.entityId)
+    }
+
+    if (isSubj == null) {
```
Comment (sugarfree1) [inline src/org/camunda/latera/bss/connectors/hid/hydra/Address.groovy]: nullable boolean is painless

Comment (dolfinus) [reply src/org/camunda/latera/bss/connectors/hid/hydra/Address.groovy]: You mean painful? (:

**Pull request merged**